src/witan/send/settings.clj

```diff
@@ -348,6 +348,33 @@
      :designation designation
      :area        area}))
 
+(defn setting-components
+  "Returns map of setting components extracted from `setting` abbreviation."
+  [setting & {:as cfg}]
+  (let [[setting estab-cat designation area]
+        (re-find (setting-components-regex cfg) setting)]
+    {:setting     setting
+     :estab-cat   estab-cat
+     :designation designation
+     :area        area}))
+
+(defn designation-component-regex
+  "Return regex pattern for extracting designation component.
+   Relies on area abbreviations not clashing with designation abbreviations.
+   Collection of areas can be specified (first supplied is used):
+   - directly via `:area-abbreviations` key
+   - via ::areas defined in settings cfg (via `(areas cfg)`)."
+  [& {designation-components-regex' ::designation-components-regex
+      area-abbreviations            :area-abbreviations ; Note deliberately NOT namespaced
+      :as                           cfg}]
+  (or designation-components-regex'
+      (let [area-abbreviations' (or area-abbreviations
+                                    (keys (areas cfg)))]
+        (re-pattern (str "(?<=_)"                                        ; positive lookbehind for an underscore
+                         "(?!(" (str/join "|" area-abbreviations') ")$)" ; negative lookahead for area-abbreviations
+                         "([^_]+)"                                       ; one or more chars except underscores
+                         )))))
```

Comment:
I feel denote.el is a good example of a better way of handling magic strings like this.

I appreciate having the regexp broken up and explained